testsuite/MDAnalysisTests/formats/test_libdcd.py

```diff
@@ -0,0 +1,520 @@
+
+
+@pytest.mark.parametrize("dcdfile, remarks", (
+    (DCD, DCD_HEADER), (DCD_NAMD_TRICLINIC, DCD_NAMD_TRICLINIC_HEADER),
+    (DCD_TRICLINIC, DCD_TRICLINIC_HEADER)))
+def test_header_remarks(dcdfile, remarks):
+    # confirm correct header remarks section reading
+    with DCDFile(dcdfile) as f:
+        assert len(f.header['remarks']) == len(remarks)
+
+
+@pytest.mark.parametrize("dcdfile, legacy_data, frames",
+                         ((DCD, legacy_DCD_ADK_coords, [5, 29]),
+                          (DCD_NAMD_TRICLINIC, legacy_DCD_NAMD_coords, [0]),
+                          (DCD_TRICLINIC, legacy_DCD_c36_coords, [1, 4])))
+def test_read_coord_values(dcdfile, legacy_data, frames):
+    # test the actual values of coordinates read in versus
+    # stored values read in by the legacy DCD handling framework
+    # to reduce repo storage burden, we only compare for a few
+    # randomly selected frames
+    legacy = np.load(legacy_data)
+    with DCDFile(dcdfile) as dcd:
+        for index, frame_num in enumerate(frames):
+            dcd.seek(frame_num)
+            actual_coords = dcd.read()[0]
+            desired_coords = legacy[index]
+            assert_array_equal(actual_coords, desired_coords)
+
+
+@pytest.mark.parametrize("dcdfile, legacy_data, frame_idx",
+                         ((DCD, legacy_DCD_ADK_coords, [5, 29]),
+                          (DCD_NAMD_TRICLINIC, legacy_DCD_NAMD_coords, [0]),
+                          (DCD_TRICLINIC, legacy_DCD_c36_coords, [1, 4])))
+def test_readframes(dcdfile, legacy_data, frame_idx):
+    legacy = np.load(legacy_data)
+    with DCDFile(dcdfile) as dcd:
+        frames = dcd.readframes()
+        xyz = frames.xyz
+        assert_equal(len(xyz), len(dcd))
+        for index, frame_num in enumerate(frame_idx):
+            assert_array_almost_equal(xyz[frame_num], legacy[index])
+
+
+def test_write_header(tmpdir):
+    # test that _write_header() can produce a very crude
+    # header for a new / empty file
+    testfile = 'test.dcd'
+    with tmpdir.as_cwd():
+        with DCDFile(testfile, 'w') as dcd:
+            dcd.write_header(
+                remarks='Crazy!',
+                natoms=22,
+                istart=12,
+                nsavc=10,
+                delta=0.02,
+                is_periodic=1)
+
+        with DCDFile(testfile) as dcd:
+            header = dcd.header
+            assert header['remarks'] == 'Crazy!'
+            assert header['natoms'] == 22
+            assert header['istart'] == 12
+            assert header['is_periodic'] == 1
+            assert header['nsavc'] == 10
+            assert np.allclose(header['delta'], .02)
```

Comment:
@orbeckst I have change the API for the header reading/writing now. It doesn't expose the CHARMM bitfield anymore. That bitfield was never stored in the dcd anyway and a  implementation detail of `readdcd.h`. Instead I'm not just giving information whether the file stores box information or not. This IMHO makes libdcd easier to use and the writing and reading of a header is now symmetric. The bitfield still exists as a property to people who want to have more information but I will document it with a warning. 

Does this satisfy you? No more bitfield manipulation that we need to document.